File: Server_Manager/Domain/Model/Response/Class/ClassResponse.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Domain.Model.Response.Class
{
    public class ClassResponse
    {
        public long Id { get; set; }
        [StringLength(100), Required]
        public string Code { get; set; }
        [StringLength(255), Required]
        public string Name { get; set; }
        [Required]
        public int Day { get; set; }
        [Required]
        public TimeOnly TimeStart { get; set; }
        [Required]
        public TimeOnly TimeEnd { get; set; }
        public long? LecturerId { get; set; }
        public long? CourseId { get; set; }
    }
}

File: chamcongspit_flutter/lib/data/models/class/ClassResponse.dart
class ClassResponse {
  int? id;
  String? code;
  String? name;
  int? day;
  String? timeStart;
  String? timeEnd;
  int? courseId;
  List<int>? studentsId;
  List<int>? lecturersId;

  ClassResponse({
    this.id,
    this.code,
    this.name,
    this.day,
    this.timeStart,
    this.timeEnd,
    this.courseId,
    this.studentsId,
    this.lecturersId,
  });

  ClassResponse.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    code = json['code'];
    name = json['name'];
    day = json['day'];
    timeStart = json['timeStart'];
    timeEnd = json['timeEnd'];
    courseId = json['courseId'];
    studentsId = json['studentsId'].cast<int>();
    lecturersId = json['lecturersId'].cast<int>();
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['code'] = code;
    data['name'] = name;
    data['day'] = day;
    data['timeStart'] = timeStart;
    data['timeEnd'] = timeEnd;
    data['courseId'] = courseId;
    data['studentsId'] = studentsId;
    data['lecturersId'] = lecturersId;
    return data;
  }

  @override
  String toString() {
    return name!;
  }
}
